system/doc/reference_manual/features.xml

```diff
@@ -0,0 +1,204 @@
+      </row>
+      <row>
+        <cell>Rejected</cell>
+        <cell>disabled</cell>
+        <cell>no</cell>
+        <cell>no</cell>
+      </row>
+      <tcaption>Feature States</tcaption>
+    </table>
+    <list>
+      <item>Being controllable means the possibility to enable or
+      disable the feature by means of compiler options and directives
+      in the file being compiled.</item>
+      <item>Being available can be seen using the
+      <c>FEATURE_AVAILABLE</c> macro.</item>
+    </list>
+  </section>
+
+  <section>
+    <title>Enabling and Disabling Features</title>
+    <p>To use a feature that is in the experimental state, it has to
+    be enabled during compilation.  This can be done in a number of
+    different ways:
+    </p>
+    <taglist>
+      <tag>Options to <c>erlc</c></tag>
+      <item>Options <seecom
+      marker="erts:erlc#enable-feature"><c>-enable-feature</c></seecom>
+      and <seecom
+      marker="erts:erlc#disable-feature"><c>-disable-feature</c></seecom>
+      can be used to enable or disable individal features.</item>
+      <tag>Compiler options</tag>
+      <item>The compiler option <seeerl
+      marker="compiler:compile#feature-option"><c>{feature,
+      &lt;feature&gt;, enable|disable}</c></seeerl> can be used either
+      as a <c>+&lt;term&gt;</c> option to <c>erlc</c>, in the options
+      argument to function in the <c>compile</c> module, or in the
+      <c>-compile(..)</c> attribute in a file.</item>
+      <tag>The feature directive</tag>
+      <item>Inside a prefix of a module, one can use a <seeerl
+      marker="compiler:compile#feature-directive"><c>-feature(&lt;feature&gt;,
+      enable|disable)</c></seeerl> directive.  This is the preferred
+      method of enabling and disabling features.</item>
+    </taglist>
+    <p>
+      Note that to load a module compiled with features enabled, the
+      corresponding features must be enabled in the runtime.  This
+      is done using options <seecom
+      marker="erts:erl#enable-feature"><c>-enable-feature</c></seecom>
+      and <seecom
+      marker="erts:erl#disable-feature"><c>-disable-feature</c></seecom>
+      to <c>erl</c>.  This is to allow the possibility to prevent
+      the use of experimental features in, e.g., production.  This
+      will catch experimental features used in both own and third
+      party components.  An active choice to use experimental
+      features must be done.
+    </p>
+  </section>
+
+  <section>
+    <title>Preprocessor Additions</title>
+    <p>
+      To allow for conditional compilation during transitioning of a
+      code base and/or trying out experimental features <seeguide
+      marker="system/reference_manual:macros#predefined-macros">feature</seeguide>
+      <c>predefined macros</c> <c>?FEATURE_AVAILABLE(Feature)</c> and
+      <c>?FEATURE_ENABLED(Feature)</c> are available.
+    </p>
+  </section>
+
+  <section>
+    <title>Information about Existing Features</title>
```

Comment:
I'm wondering if there should be a place I'm the documentation where all the experimental features are listed? Maybe we don't need to keep the permanent or rejected features there, but the experimental and accepted might be nice to have? If nothing else just links to the actual documentation of the features, i.e. to the maybe feature now.

Reply:
I agree.  Where?  In the documentation for `maybe` it just says that it is a experimental, but it might be a good idea to have a more "formal" marking of it as well.  Just having _since_ is not really enough.  I added a link to this general section from it, though.

Reply:
The best place I can think of is at the end of this page in the reference manual.

Reply:
Added now, with a reference to the sole existing feature.  Ideally, this should be generated automagically for each release.  It could be done using the functions in `erl_features`, but not enough time for that now.

Reply:
Great! and yes, it should indeed be automated.